extension/src/popup/helpers/soroban.ts

```diff
@@ -209,3 +211,115 @@ export const isContractId = (contractId: string) => {
     return false;
   }
 };
+
+interface InvocationTree {
+  type: string;
+  args: any;
+  invocations: InvocationTree[];
+}
+
+export function buildInvocationTree(root: xdr.SorobanAuthorizedInvocation) {
```

Comment:
This is temporarily pulled from stellar-sdk, I need to open a PR to parse the fnName to a string but there's urgency to clear this work so I pulled it into this project for the meantime.